app/models/mixins/live_metrics_mixin.rb

```diff
@@ -33,4 +38,15 @@ def adjust_timestamps(firsts, lasts, interval_name)
       [first, last]
     end
   end
+
+  module ClassMethods
+    def supported_metrics
+      @supported_metrics ||= load_supported_metrics
+    end
+
+    def load_supported_metrics
```

Comment:
I re-used the same pattern in MiQ for load a YAML.
In case of error, supported_metrics variable is an empty hash.